include/Socket.h

```diff
@@ -3,20 +3,26 @@
 
 #include <string>
 #include <unistd.h>
+#include <vector>
+
+#define MAX_BUFFER_SIZE 1024
```

Comment:
I suggest to make it a private static int member to prevent any conflicts in macro names.